middleware/authenticateUser.js

```diff
@@ -1,26 +1,55 @@
-const { auth } = require("../firebase");
+const jwt = require("jsonwebtoken");
 const config = require("../config");
 
-const authenticateUser = async (req, res, next) => {
+// Function to generate JWT token
+const generateToken = (payload) => {
+  return jwt.sign(payload, config.get("jwt_secret"), { expiresIn: "1h" });
+};
+
+// Function to verify JWT token
+const verifyToken = (token) => {
   try {
-    if (config.get("env") === "test_remote") next();
-    const headerToken = req.headers.authorization;
-    if (!headerToken) {
-      return res.status(401).send({ message: "No token provided" });
-    }
-
-    if (headerToken && headerToken.split(" ")[0] !== "Bearer") {
-      res.status(401).send({ message: "Invalid token" });
-    }
-
-    let token = headerToken.split(" ")[1];
-    await auth.verifyIdToken(token);
-
-    next();
-  } catch (error) {
-    req.log.error(error);
-    res.status(403).send({ message: "Could not authorize" });
+    return jwt.verify(token, config.get("jwt_secret"));
+  } catch (err) {
+    console.error("Token verification failed:", err);
+    return null;
+  }
+};
+
+// Example usage in authenticateUser function
+const authenticateUser = (req, res) => {
+  const { username, password } = req.body;
+
+  // Replace this with your user authentication logic
+  const user = authenticateWithDatabase(username, password);
+
+  if (user) {
+    const token = generateToken({ id: user.id, username: user.username });
+    res.json({ token });
+  } else {
+    res.status(401).json({ message: "Authentication failed" });
```

Comment:
these example should be part of the README or docs (which doesn't exist right now) so we can have it in the user controller instead